indico/modules/events/registration/controllers/management/fields.py

```diff
@@ -49,6 +50,20 @@ def _check_input_type(self, input_type, **kwargs):
         if input_type not in get_field_types():
             raise ValidationError('Invalid field type')
 
+    @validates_schema(skip_on_field_errors=True)
+    def _validate_retention_period(self, data, **kwargs):
+        retention_period = data.get('retention_period')
+        field = self.context['field']
+
+        if retention_period is not None:
+            if field.type == RegistrationFormItemType.text:
+                raise ValidationError('Cannot set retention period for static labels', 'retention_period')
+            elif (
+                field.type == RegistrationFormItemType.field_pd and
+                field.personal_data_type.is_required
+            ):
```

Comment:
I think this fits in a single line now:

```suggestion
            elif field.type == RegistrationFormItemType.field_pd and field.personal_data_type.is_required:
```